From akpm@linux-foundation.org  Wed Mar  5 11:42:03 2008
From: akpm@linux-foundation.org
Date: Tue, 04 Mar 2008 15:25:11 -0800
Subject: USB: usbaudio: handle kcalloc failure
To: greg@kroah.com
Cc: linux-usb@vger.kernel.org, akpm@linux-foundation.org, jim@meyering.net, meyering@redhat.com
Message-ID: <200803042325.m24NPBOa003214@imap1.linux-foundation.org>


From: Jim Meyering <jim@meyering.net>

sound/usb/usbaudio.c (check_hw_params_convention): Handle kcalloc failure.

Signed-off-by: Jim Meyering <meyering@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 sound/usb/usbaudio.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/sound/usb/usbaudio.c
+++ b/sound/usb/usbaudio.c
@@ -1762,6 +1762,8 @@ static int check_hw_params_convention(st
 
 	channels = kcalloc(MAX_MASK, sizeof(u32), GFP_KERNEL);
 	rates = kcalloc(MAX_MASK, sizeof(u32), GFP_KERNEL);
+	if (!channels || !rates)
+		goto __out;
 
 	list_for_each(p, &subs->fmt_list) {
 		struct audioformat *f;
